spec/models/oidc_request_spec.rb

```
@@ -0,0 +1,342 @@
+                                         .and_return(provider.merge('scopes' => nil))
+      allow(client).to receive(:authorization_uri)
+                         .with(hash_including(scope: %w[openid email profile], code_challenge_method: 'S256'))
+                         .and_return('https://accounts.google.com/o/oauth2/v2/auth?scope=openid+email+profile')
+
+      described_class.authorization_uri_for!(provider_key: 'google-ncsu', username: 'oidcuser')
+    end
+
+    it 'raises when creating a request with a duplicate state' do
+      create_request(state: 'duplicate-state')
+
+      expect {
```

RuboCop (1.86.1)

[convention] Avoid using `{...}` for multi-line blocks.

(Style/BlockDelimiters)